core/node/node_stats_tracker.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2022 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package node
+
+import "github.com/pkg/errors"
+
+// MonitoringAgentStatuses a object represent a [service_type][status]amount of statuses for each service type.
+type MonitoringAgentStatuses map[string]map[string]int
+
+// ProviderStatuses should return provider statuses from monitoring agent
+type ProviderStatuses func(providerID string) (MonitoringAgentStatuses, error)
+
+// ProviderSessionsList should return provider sessions list
+type ProviderSessionsList func(providerID, rangeTime string) (SessionsList, error)
+
+// StatsTracker tracks metrics for service
+type StatsTracker struct {
+	providerStatuses     ProviderStatuses
+	providerSessionsList ProviderSessionsList
+	currentIdentity      currentIdentity
+}
+
+// NewNodeStatsTracker constructor
+func NewNodeStatsTracker(
+	providerStatuses ProviderStatuses,
+	providerSessions ProviderSessionsList,
+	currentIdentity currentIdentity,
+) *StatsTracker {
+	mat := &StatsTracker{
+		providerStatuses:     providerStatuses,
+		providerSessionsList: providerSessions,
+		currentIdentity:      currentIdentity,
+	}
+
+	return mat
+}
+
+// Statuses retrieves and resolved monitoring status from quality oracle
+func (m *StatsTracker) Statuses() (MonitoringAgentStatuses, error) {
+	id, ok := m.currentIdentity.GetUnlockedIdentity()
+	if ok {
+		return m.providerStatuses(id.Address)
+	}
+
+	return MonitoringAgentStatuses{}, errors.New("identity not found")
+}
+
+// SessionItem represents information about session monitoring metrics.
+type SessionItem struct {
```

Comment:
Our PO also had ideas about `Service Type` of given session 

![image](https://user-images.githubusercontent.com/20511486/184068009-a17898a0-7d03-4b45-8259-16665e03e034.png)

Reply:
yep, this is needed since we are close to having many service types.